### PR TITLE
Adding .NET Core binary serialization doc

### PR DESCRIPTION
**Pull request by ViktorHofer**
Relates to https://github.com/dotnet/docs/issues/2770

Adding a doc which describes the differences between .NET Core and .NET Framework in regards to binary serialization.

@mairaw I haven't spent time to figure out how to add the doc into the navigation pane and what else is required. You could either tell me or commit directly into the PR here.

### CODE REVIEW

**Comment by mairaw** [comment]
I can finish the PR for you. I'll put myself as the ms.author so that I'm responsible for looking at feedback, etc. but leave you as the author of the article. Is that ok?

**Comment by ViktorHofer** [comment]
Perfect. Thanks Maira!

cc @danmosemsft

**Comment by mairaw** [comment]
@ViktorHofer please take a look at what I did:
- updated metadata 
- renamed file to have important keywords
- transformed the list in active links

What do you think?

Also, I can't seem to find TreeSet. Is this already available?

Thanks.

**Comment by ViktorHofer** [comment]
> What do you think?
LGTM. Thanks for the changes. I think the active link list makes sense :)

> Also, I can't seem to find TreeSet. Is this already available?
https://github.com/dotnet/corefx/blob/master/src/System.Collections/src/System/Collections/Generic/SortedDictionary.cs#L960
Maybe you can't find it because it is marked as obsolete.

File: docs/standard/serialization/binary-serialization-core.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+- <xref:System.ArraySegment%601?displayProperty=fullName>   
+- <xref:System.Attribute?displayProperty=fullName>   
+- <xref:System.Boolean?displayProperty=fullName>   
+- <xref:System.Byte?displayProperty=fullName>   
+- <xref:System.Char?displayProperty=fullName>   
+- <xref:System.Collections.ArrayList?displayProperty=fullName>   
+- <xref:System.Collections.BitArray?displayProperty=fullName>   
+- <xref:System.Collections.Comparer?displayProperty=fullName>   
+- <xref:System.Collections.DictionaryEntry?displayProperty=fullName>   
+- <xref:System.Collections.DictionaryNode?displayProperty=fullName>
```
Comment (mairaw) [inline docs/standard/serialization/binary-serialization-core.md]: I can see this in the source (https://github.com/dotnet/corefx/blob/release/2.0.0/src/System.Collections.Specialized/src/System/Collections/Specialized/ListDictionary.cs#L496) but can't see this on the docs/reference assemblies either

File: docs/standard/serialization/binary-serialization-core.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+- <xref:System.Attribute?displayProperty=fullName>   
+- <xref:System.Boolean?displayProperty=fullName>   
+- <xref:System.Byte?displayProperty=fullName>   
+- <xref:System.Char?displayProperty=fullName>   
+- <xref:System.Collections.ArrayList?displayProperty=fullName>   
+- <xref:System.Collections.BitArray?displayProperty=fullName>   
+- <xref:System.Collections.Comparer?displayProperty=fullName>   
+- <xref:System.Collections.DictionaryEntry?displayProperty=fullName>   
+- <xref:System.Collections.DictionaryNode?displayProperty=fullName>   
+- <xref:System.Collections.Generic.ByteEqualityComparer?displayProperty=fullName>
```
Comment (mairaw) [inline docs/standard/serialization/binary-serialization-core.md]: this is internal right?

File: docs/standard/serialization/binary-serialization-core.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+- <xref:System.Globalization.SortVersion?displayProperty=fullName>   
+- <xref:System.Guid?displayProperty=fullName>   
+- <xref:System.Int16?displayProperty=fullName>   
+- <xref:System.Int32?displayProperty=fullName>   
+- <xref:System.Int64?displayProperty=fullName>   
+- <xref:System.IntPtr?displayProperty=fullName>   
+- <xref:System.Net.Cookie?displayProperty=fullName>   
+- <xref:System.Net.CookieCollection?displayProperty=fullName>   
+- <xref:System.Net.CookieContainer?displayProperty=fullName>   
+- <xref:System.Net.PathList?displayProperty=fullName>
```
Comment (mairaw) [inline docs/standard/serialization/binary-serialization-core.md]: another internal correct?

File: docs/standard/serialization/binary-serialization-core.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+- <xref:System.Uri?displayProperty=fullName>   
+- <xref:System.ValueTuple?displayProperty=fullName>   
+- <xref:System.ValueType?displayProperty=fullName>   
+- <xref:System.Version?displayProperty=fullName>   
+- <xref:System.WeakReference?displayProperty=fullName>   
+- <xref:System.WeakReference%601?displayProperty=fullName>   
+
+## Remarks
+
+As the nature of binary serialization allows the modification of private members inside an object and therefore changing the state of it, other serialization frameworks like JSON.NET which operate on the public API surface are recommended.
```
Comment (mairaw) [inline docs/standard/serialization/binary-serialization-core.md]: this is applicable to any binary serialization, not only in .net core right? Should this go to the main topic perhaps? I'll also try to rework this sentence. It's a bit hard to parse.

**Review by mairaw — COMMENTED** [review]
You can see the preview topic here: https://review.docs.microsoft.com/en-us/dotnet/standard/serialization/binary-serialization-core?branch=pr-en-us-2841

File: docs/standard/serialization/binary-serialization-core.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+author: "ViktorHofer"
+ms.author: "mairaw"
+---
+# Supported types for binary serialization in .NET Core
+
+ .NET Core supports binary serialization with a subset of types. You can see the list of supported types in the [next section](#serializable-types). The defined set of types are guaranteed to be serializable between .NET Framework 4.5.1 and later versions and .NET Core 2.0 and later versions. Other .NET implementations, such as Mono, aren't officially supported but should also be working.
+
+## Serializable types
+
+- <xref:System.AdjustmentRule?displayProperty=fullName>
```
Comment (ViktorHofer) [inline docs/standard/serialization/binary-serialization-core.md]: Oooh my, yes you are correct. My tool which generated the list stripped it

File: docs/standard/serialization/binary-serialization-core.md (hunk 1 above)
Comment (ViktorHofer) [inline docs/standard/serialization/binary-serialization-core.md]: Correct. It isn't exposed in the ref assemblies therefore we can remove the type from the list as we don't want people compiling against the src assemblies. The type is public in the src because of .NET Native Toolchain. I missed that. Thanks

File: docs/standard/serialization/binary-serialization-core.md (hunk 2 above)
Comment (ViktorHofer) [inline docs/standard/serialization/binary-serialization-core.md]: Yes remove please.

File: docs/standard/serialization/binary-serialization-core.md (hunk 3 above)
Comment (ViktorHofer) [inline docs/standard/serialization/binary-serialization-core.md]: Yes remove please.

File: docs/standard/serialization/binary-serialization-core.md (hunk 4 above)
Comment (ViktorHofer) [inline docs/standard/serialization/binary-serialization-core.md]: Jep, we can put that into the general section. It's important for us to warn users to think about using different technologies when it comes to new solutions.

Comment (mairaw) [inline docs/standard/serialization/binary-serialization-core.md]: Sounds good. I'll work on your feedback. Appreciate it.

Few remaining questions and then I can finish the edits and merge:
- Should I also remove TreeSet or keep that one?
- I was thinking that maybe the content of this file could move into the main binary serialization. Thoughts on that?

Comment (ViktorHofer) [inline docs/standard/serialization/binary-serialization-core.md]: Yes, you can remove TreeSet as it is non-existent in the ref assemblies.

Content of this file? You mean the whole file? I'm okay with that if it remains titled as NET Core specific...

Thanks Mairaw for your work!!

Comment (mairaw) [inline docs/standard/serialization/binary-serialization-core.md]: My suggestion was to move as a subsection with that title within https://github.com/dotnet/docs/blob/master/docs/standard/serialization/binary-serialization.md. Do you need an individual topic for this @ViktorHofer?

No worries! My pleasure.

Comment (ViktorHofer) [inline docs/standard/serialization/binary-serialization-core.md]: Jep, I'm fine with that. Thanks

**Comment by mairaw** [comment]
Ok, this is done. As soon CI is green and things look good, I'll merge, unless I hear otherwise. Thanks!

**Review by JRAlexander — APPROVED** [review]
LGTM
